crates/proto/src/xfer/dnssec_dns_handle/mod.rs

```diff
@@ -385,10 +394,16 @@ where
                 debug!("verified: {name} record_type: {record_type}",);
                 (proof, adjusted_ttl)
             }
-            Err(ProofError { proof, kind }) => {
-                debug!("failed to verify: {name} record_type: {record_type}: {kind}",);
-                (proof, None)
-            }
+            Err(ProofError { proof, kind }) => match kind {
+                ProofErrorKind::DsResponseNsec { .. } => {
+                    debug!("verified insecure {name}/{record_type}");
+                    (proof, None)
+                }
+                _ => {
+                    debug!("failed to verify: {name} record_type: {record_type}: {kind}",);
+                    (proof, None)
+                }
+            },
```

Comment:
Suggest making it clear that the return value does not depend on `kind` by taking that out of the `match` expression.